### DISCUSSION TITLE
fast dds can support android platform and how to compile fast dds to suport andriod?

### DISCUSSION THREAD

**Discussion by longqionggaccount** [eProsima/Fast-DDS#4910] [Q&A]
fast dds can support android platform and how to compile fast dds to suport andriod?

**Comment by JesusPoderoso** [comment]
Hi @longqionggaccount, thanks for using Fast DDS.
Refer to [this section of the documentation](https://fast-dds.docs.eprosima.com/en/latest/notes/versions.html#build-system-dependencies) to check the compatible SDK versions per Fast DDS version. Note that Android OS is [Tier 3](https://fast-dds.docs.eprosima.com/en/latest/notes/versions.html#platform-support).
If it helps, you can find some `CMake arguments` of interest related to Android in [this section of the documentation](https://fast-dds.docs.eprosima.com/en/latest/installation/configuration/cmake_options.html).
If you need more help in relation to that, do not hesitate contacting our commercial support: [support@eprosima.com](mailto:support@eprosima.com)

**Reply by andrei-ng** [reply]
Hi @longqionggaccount 
I have been using this https://github.com/eyr1n/fastdds-prefab with an Android project and works like a charm.

**Reply by CNXiaoZhi** [reply]
> Hi @longqionggaccount I have been using this https://github.com/eyr1n/fastdds-prefab with an Android project and works like a charm.

oh ! it realy helpful，thanks!!!!!!

**Comment by longqionggaccount** [comment]
邮件已收到，谢谢！

**Comment by CHYou2Sef** [comment]
Hi there! 

Yes, Fast DDS fully supports the Android platform. It is designed to be portable and can be cross-compiled for Android using the Android NDK (Native Development Kit).

To compile it for Android, you will first need to compile its main dependency, Fast CDR, and then compile Fast DDS using CMake with the Android toolchain file.

Here is the standard CMake procedure to cross-compile it:

1. Prerequisites: Ensure you have the Android NDK installed and the path set in your environment (e.g., $ANDROID_NDK).

2. Compile command: You can use the following CMake command, adjusting the API level and ABI to match your target device:

Bash

cmake -DCMAKE_SYSTEM_NAME=Android \
      -DCMAKE_SYSTEM_VERSION=24 \
      -DCMAKE_ANDROID_ARCH_ABI=arm64-v8a \
      -DCMAKE_ANDROID_NDK=$ANDROID_NDK \
      -DCMAKE_TOOLCHAIN_FILE=$ANDROID_NDK/build/cmake/android.toolchain.cmake \
      -DANDROID_PLATFORM=android-24 \
      ..
make
Note: Ensure that you cross-compile and install Fast CDR using the same flags before compiling Fast DDS, so that CMake can locate the package.

I hope this helps you get your build running! Let me know if you run into specific linker errors.

**Comment by longqionggaccount** [comment]
邮件已收到，谢谢！
